networkx/algorithms/tests/test_dag.py

```diff
@@ -60,6 +60,23 @@ def test_unorderable_nodes(self):
         # this will raise NotImplementedError when nodes need to be ordered
         nx.dag_longest_path(G)
 
+    def test_multigraph_unweighted(self):
+        edges = [(1, 2), (2, 3), (2, 3), (3, 4), (4, 5), (1, 3), (1, 5), (3, 5)]
+        G = nx.MultiDiGraph(edges)
+        assert nx.dag_longest_path(G) == [1, 2, 3, 4, 5]
+
+    def test_multigraph_weighted(self):
+        G = nx.MultiDiGraph()
+        edges = [
+            (1, 2, 2),
+            (2, 3, 2),
+            (1, 3, 1),
+            (1, 3, 5),
+            (1, 3, 2),
+        ]
+        G.add_weighted_edges_from(edges)
+        assert nx.dag_longest_path(G) == [1, 3]
```

Comment:
The tests look great, thanks! It might also be worth adding a basic test for the `default_weight` too, maybe something like:

```suggestion
        assert nx.dag_longest_path(G) == [1, 3]
        
    def test_multigraph_weighted_default_weight(self):
        G = nx.MultiDiGraph([(1, 2), (2, 3)])  # Unweighted edges
        G.add_weighted_edges_from([(1, 3, 1), (1, 3, 5), (1, 3, 2)])
        
        # Default value for default weight is 0
        assert nx.dag_longest_path(G) == [1, 3]
        assert nx.dag_longest_path(G, default_weight=3) == [1, 2, 3]
```

... with a similar added test for the `dag_longest_path_length` function.